examples/repl.rs

```diff
@@ -63,98 +65,97 @@ fn prepare_home_dir() -> PathBuf {
-    let tree = database
-        .open_tree(matches.value_of("wallet").unwrap())
-        .unwrap();
+    let tree = database.open_tree(cli_opt.wallet).unwrap();
     debug!("database opened successfully");
 
-    let config = match matches.value_of("esplora") {
+    let config = match cli_opt.esplora {
         Some(base_url) => AnyBlockchainConfig::Esplora(EsploraBlockchainConfig {
             base_url: base_url.to_string(),
-            concurrency: matches
-                .value_of("esplora_concurrency")
-                .and_then(|v| v.parse::<u8>().ok()),
+            concurrency: Some(cli_opt.esplora_concurrency),
         }),
         None => AnyBlockchainConfig::Electrum(ElectrumBlockchainConfig {
-            url: matches.value_of("server").unwrap().to_string(),
-            socks5: matches.value_of("proxy").map(ToString::to_string),
+            url: cli_opt.electrum,
+            socks5: cli_opt.proxy,
             retry: 10,
             timeout: 10,
         }),
     };
-    let wallet = Arc::new(
-        Wallet::new(
-            descriptor,
-            change_descriptor,
-            network,
-            tree,
-            AnyBlockchain::from_config(&config).unwrap(),
-        )
-        .unwrap(),
-    );
-
-    if let Some(_sub_matches) = matches.subcommand_matches("repl") {
-        let mut rl = Editor::<()>::new();
-
-        // if rl.load_history("history.txt").is_err() {
-        //     println!("No previous history.");
-        // }
-
-        loop {
-            let readline = rl.readline(">> ");
-            match readline {
-                Ok(line) => {
-                    if line.trim() == "" {
-                        continue;
-                    }
 
-                    rl.add_history_entry(line.as_str());
-                    let matches = repl_app.get_matches_from_safe_borrow(line.split(" "));
-                    if let Err(err) = matches {
-                        println!("{}", err.message);
-                        continue;
+    let wallet = Wallet::new(
+        descriptor,
+        change_descriptor,
+        network,
+        tree,
+        AnyBlockchain::from_config(&config).unwrap(),
+    )
+    .unwrap();
+
+    let wallet = Arc::new(wallet);
+
+    match cli_opt.subcommand {
+        WalletSubCommand::Other(external) if external.contains(&"repl".to_string()) => {
+            let mut rl = Editor::<()>::new();
+
+            // if rl.load_history("history.txt").is_err() {
+            //     println!("No previous history.");
+            // }
+
+            loop {
+                let readline = rl.readline(">> ");
+                match readline {
+                    Ok(line) => {
+                        if line.trim() == "" {
+                            continue;
+                        }
+                        rl.add_history_entry(line.as_str());
+                        let mut repl_args: Vec<&str> = vec![" "];
```

Comment:
I think you can avoid having to do this if you find a way to add the [`NoBinaryName`](https://docs.rs/clap/2.33.3/clap/enum.AppSettings.html#variant.NoBinaryName) flag. IIRC we were also doing this before somewhere here in the repl example.

That's useful when parsing something like a Vec/iterator of arguments.. basically whenever you are parsing a custom list of arguments rather than the command line from the OS. Not sure if it makes sense to have that as a default, but I would at least write somewhere in the docs that it can be done

Reply:
Unfortunately if I add the `NoBinaryName` setting to `WalletSubCommand` it works for repl but breaks when using it for cli args. As a work around I created a new `ReplOpt` struct inr `repl.rs` with just a `WalletSubCommand` and setting `NoBinaryName` and this works. I also added some explaination in the docs.

Reply:
https://github.com/bitcoindevkit/bdk/pull/208/commits/364b47bfcb4a6f2ce4c9a6f34be92dbe4a301a73#diff-41438e9011cd34803a435657ff5534e6ff5a053c7b75b2d08fdcdbe7b309e78dR47